### DISCUSSION TITLE
When searching support matching of ASCII characters and stylized Unicode counterparts

### DISCUSSION THREAD

**Discussion by bartkl** [jcorporation/myMPD#878] [Ideas]
myMPD version: 10.0.3

**Describe the bug**

When searching, basic ASCII characters do not match stylized Unicode counterparts. For example, searching for "death's" doesn't match "death’s" (note the different apostrophe character).

I get my metadata from MusicBrainz, where such stylization is done in many different flavors (dashes, double quotes, horizontal ellipses, ...). It would be nice to support this.

Thanks!

**To Reproduce**

See example above.

**Expected behavior**

Basic ASCII characters are easy to type, which is a requirement for a usable search. Related Unicode characters should match these in my opinion.

**Screenshots**

n/a

**Server plattform (please complete the following information):**
 - OS: Raspbian
 - OS version 10
 - Plattform: armv7l
 - MPD version: 0.23.2

**Client plattform (please complete the following information):**
 - Device: Lenovo ThinkPad laptop
 - OS: Pop!_OS 22.04 LTS (Ubuntu based)
 - Browser: Firefox
 - Browser version: 105.0.3 (64-bit)

**Debug logs (please attach if it can be usefull)**

n/a
 
**Configuration (please attach if it can be usefull)**

n/a

**Additional context**

n/a

**Comment by jcorporation** [comment]
Song search is done by MPD, that is a feature request for MPD. I avoid reimplementing MPD features.

**Reply by bartkl** [reply]
I understand that decision.

It's surprising to see that, yes indeed, MPD itself is responsible for this. I think I'll open an issue at the MPD repo. It seems to me a rather reasonable requirement from a user POV.

**Reply by bartkl** [reply]
Update: I opened an issue [here](https://github.com/MusicPlayerDaemon/MPD/issues/1669)
